### DISCUSSION TITLE
How to find nested matches in a JSON column?

### DISCUSSION THREAD

**Discussion by mmmmmrob** [prisma/prisma#13810] [Q&A]
Hi,

We have a table of nested JSON trees, the trees are arbitrary depth, nodes in the tree have UUIDs.

The trees are a jsonb column in a postgres table.

```json
[
  {
    "id": "44631a5b-cbec-423d-a259-b164dda4a8c6",
    "children": [
      {
        "id": "67e5918c-2c35-4722-8eaa-c5db7c91a92e",
        "children": []
      }
    ]
  }
]
```

We have lots of these trees and the depth varies. We need to find trees that contain a node with any of a set of UUIDs.

```json
["44631a5b-cbec-423d-a259-b164dda4a8c6", "…", "…", …]
```

```js
prisma.tree.findMany({ where: ??? })
```

How can we query something like `$.**.id in ["44631a5b-cbec-423d-a259-b164dda4a8c6", "…", "…", …]`?

i.e. get back all trees where any node is one of these UUIDs.

Thanks.

🙏

**Comment by austincrim** [comment]
Hey @mmmmmrob 👋!

Per [this docs page](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filtering-on-object-key-value-inside-array), I don't think this is possible when using Postgres JSON and Prisma.

You could try using a raw query and Postgres' native JSONB query operators.

**Reply by mmmmmrob** [reply]
We currently have a `queryRawUnsafe` in use and are trying to move away from it.

We're unable to use `queryRaw` because it does not correctly encode the json path when passed as a parameter.

Code for that is:

```sql
const jsonPath = nodeIds.map((id) => `$.**.id == "${id}"`).join(' || ')

const trees = await prisma.$queryRawUnsafe(`
      SELECT *
      FROM trees
      WHERE jsonb_path_match(nodes, '${jsonPath}')
    `)
```

**Reply by austincrim** [reply]
I think what you have there is your best bet for now.

Would you mind opening a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) with more details about what you would like to see in the future?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
